### PR TITLE
Add findLast and findLastIndex.

### PR DESCRIPTION
**Pull request by pushplay**

### CODE REVIEW

File: feature-deep-dive/rule-syntax.md [hunk 1]
```diff
@@ -214,6 +214,30 @@ findIndex([1, 2, 3], x => x % 2 == 0) → 1
...
 ```
 
+### findLast
+
+`findLast(list: any[], condition: (listElement: any) => boolean) → any`
+
+Get the last element of a list that meets the given condition.  Returns null if the condition is never met.
+
+```javascript
+find([1, 2, 3], x => x % 2 == 0) → 2
```
Comment (Tim-Jordison) [inline feature-deep-dive/rule-syntax.md]: Typo. Should be `findLast`.

File: feature-deep-dive/rule-syntax.md [hunk 2]
```diff
@@ -214,6 +214,30 @@ findIndex([1, 2, 3], x => x % 2 == 0) → 1
...
+```
+
+### findLastIndex
+
+`findLastIndex(list: any[], condition: (listElement: any) => boolean) → number`
+
+Get the index of the last element of a list that meets a given condition.  Returns -1 if the condition is never met.
+
+```javascript
+findIndex([1, 2, 3], x => x % 2 == 0) → 1
```
Comment (Tim-Jordison) [inline feature-deep-dive/rule-syntax.md]: Typo. These should be `findLastIndex`.

File: feature-deep-dive/rule-syntax.md (hunk 1 above)
Comment (pushplay) [reply feature-deep-dive/rule-syntax.md]: Good catch.

File: feature-deep-dive/rule-syntax.md [hunk 3]
```diff
@@ -450,7 +450,7 @@ values({itemId: '33bbb2bf-c270-41d9-ab42-9eeba99fa69c', size: 'medium', quantity
 
 Identifiers are used in variable names, map dot notation and lambda parameter names.
 
-An identifier must start with a letter (a-z, A_Z), underscore (_), or dollar sign ($).  Following characters can also be digits (0-9).  Identifiers are case sensitive.
+An identifier must start with a letter (a-z, A-Z), underscore (_), or dollar sign ($).  Following characters can also be digits (0-9).  Identifiers are case sensitive.
```
Comment (Tim-Jordison) [inline feature-deep-dive/rule-syntax.md]: 👍

**Pull request merged**